app/reducers/articles.js

```diff
@@ -24,19 +26,17 @@ export type ArticleEntity = {
   youtubeUrl: string
 };
 
-function mutateArticle(state: any, action: any) {
-  switch (action.type) {
-    case Article.DELETE.SUCCESS: {
-      const { articleId } = action.meta;
-      return {
-        ...state,
-        items: state.items.filter(id => id !== articleId)
-      };
+type State = any;
+
+const mutateArticle = produce(
+  (newState: State, action: any): void => {
```

Comment:
Why use any for the action instead of typing it properly?

Reply:
I figured the effort to type actions properly would be a bit wasted since we are doing a ts rewrite.  
I tried using https://github.com/webkom/lego-webapp/blob/master/app/types.js#L74 but it yielded lots of errors when trying to access parts of action.payload

Reply:
Ahh

Payload is however optional, aka. can be null. Our reducers mostly doesn't care tho, so using the type can help avoid nullptr exceptions

Reply:
What payload is/can be is really dependent on action.type. Not sure if this can be expressed with flow or typescript. 

It looks like typescript can handle this really well
https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions
We will need to define a type for each action type though, but I think that makes sense if we are to get any benefit from typing.

Reply:
Yeah, I looked into this as well, there seem to be well-suited methods for typing reducers.
https://www.matthewgerstman.com/tech/ts-tricks-type-safe-reducers/